packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -576,19 +573,19 @@ private Display getDefaultDisplay() {
   }
 
   /** Starts a background thread and its {@link Handler}. */
-  @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
   public void startBackgroundThread() {
-    backgroundHandlerThread = new HandlerThread("CameraBackground");
-    try {
-      backgroundHandlerThread.start();
-    } catch (IllegalThreadStateException e) {
-      // Ignore exception in case the thread has already started.
+    if (backgroundHandlerThread == null) {
```

Comment:
Could we add a lifecycle test to avoid re-involved with the same issue?

Reply:
There is a unit-test which tests for the `if` statement. However I am not sure how we could add a test to verify that we don't add implementation of listening to lifecycle events in the future. 

Do you have some insights on how this could be accomplished?

Reply:
Add an observer `WidgetsBindingObserver` in a Flutter App, then override the `didChangeAppLifecycleState` might be a solution.

Reply:
Capturing from a separate discussion, verifying that the core classes (`Camera`, `CameraPlugin`) are not `instanceof LifecycleObserver` seems like a possible approach, even if it's not foolproof (e.g., a composed object could implement it)

Reply:
alternatively, it could just return earlier.

Reply:
```java
if (backgroundHandlerThread  != null) {
  return;
}
```